gateway/ln-gateway/src/actor.rs

```diff
@@ -9,24 +9,34 @@ use fedimint_server::modules::{
 };
 use mint_client::{GatewayClient, PaymentParameters};
 use rand::{CryptoRng, RngCore};
-use tracing::{debug, instrument, warn};
+use tracing::{debug, info, instrument, warn};
 
-use crate::{ln::LnRpc, LnGatewayError, Result};
+use crate::{ln::LnRpc, utils::retry, LnGatewayError, Result};
 
 pub struct GatewayActor {
     client: Arc<GatewayClient>,
 }
 
 impl GatewayActor {
     pub async fn new(client: Arc<GatewayClient>) -> Result<Self> {
-        // Regster gateway actor with federation
-        // FIXME: This call is critically dependent on the federation being up and running.
-        // We should either use a retry strategy, OR register federations on the gateway at runtime
-        // as proposed in https://github.com/fedimint/fedimint/issues/699
-        client
-            .register_with_federation(client.config().into())
-            .await
-            .expect("Failed to register with federation");
+        // Retry regster gateway federation client with federation
+        match retry(
```

Comment:
// TODO: Retry with backoff

Reply:
I wouldn't bother with backoff. Backoff is more important for N:1 cases where many clients could effectively DDoS server. We don't expect to have many gateways, so it doesn't seem necessary.

I would worry more about the total timeout. What if connection takes 5 minutes to timeout on TCP level due to firewall blocking. Then the whole thing can basically hang `retry_times * 5 minutes".

Reply:
> I wouldn't bother with backoff. Backoff is more important for N:1 cases where many clients could effectively DDoS server. We don't expect to have many gateways, so it doesn't seem necessary.

Makes sense

> I would worry more about the total timeout. What if connection takes 5 minutes to timeout on TCP level due to firewall blocking. Then the whole thing can basically hang `retry_times * 5 minutes".**

makes me think `retry_times` and duration between retries could be heuristics configured by the gateway admin. I imagine this would give  admin the final say on when to 'give up' on a federation - after trying to register as a service provider x-times and failed